pkg/trace/stats/quantile/ddsketch_converter.go

```diff
@@ -0,0 +1,180 @@
+package quantile
+
+import (
+	"errors"
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/sketches-go/ddsketch/mapping"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/gogo/protobuf/proto"
+	"sort"
+)
+
+type ddSketch struct {
+	contiguousBins []float64
+	bins map[int32]float64
+	offset int
+	zeros int
+	mapping mapping.IndexMapping
+}
+
+// get returns the count for a given index.
+func (s *ddSketch) get(index int) (count int) {
+	if index >= s.offset && index < s.offset + len(s.contiguousBins) {
+		count = int(s.contiguousBins[index-s.offset])
+	}
+	if c, ok := s.bins[int32(index)]; ok {
+		count += int(c)
+	}
+	return count
+}
+
+func (s *ddSketch) maxSize() int {
+	return len(s.bins) + len(s.contiguousBins)
+}
+
+func getIndexes(s1 ddSketch, s2 ddSketch) []int {
```

Comment:
there is a lot of logic here to make sure we don't put the same index twice in the array.

Another way to do it would be to add a condition `if index == previousIndex { continue }` in the main loop. It might be easier. What do you think?